app/src/main/java/com/android/sample/model/activity/ActivitiesRepositoryFirestore.kt

```diff
@@ -126,7 +126,8 @@ open class ActivitiesRepositoryFirestore @Inject constructor(private val db: Fir
         status = ActivityStatus.valueOf(data["status"] as? String ?: "ACTIVE"),
         type = activityType,
         participants = participants,
-        comments = comments)
+        comments = comments,
+        category = Category.valueOf(data["category"] as? String ?: "SPORT"))
```

Comment:
Sure thing to do. You are converting the comments from the repository data. Null cases are well managed. However, why choose SPORT as the default option ?